### PR TITLE
Align documentation text with README, fixing errors along the way

### PR DESCRIPTION
**Pull request by ivan-borovets**
I aligned the documentation text with the corresponding sections in the README and fixed some minor errors along the way without altering the original wording.

A few questions remain:
1. What about `provide_all`?
Should we mention it in the `README` or in `concepts.rst`?

2. Should we include links in `index.rst` to the corresponding docs sections? If so, how?

### CODE REVIEW

File: docs/concepts.rst [hunk 1]
```diff
@@ -9,55 +9,71 @@ Let's start with some terms.
...
-Some dependencies can live while application is running, others are created and destroyed on each request. In more rare cases you need more short-lived objects. You set a scope for your dependency when you configure how to create it.
+Some dependencies live for the entire application lifetime, while others are created and destroyed with each request.
+In more rare cases, you need more short-lived objects.
+You set a scope for each dependency when you configure how it is created.
 
-Standard scopes are (excluding skipped):
+Standard scopes are (with some skipped):
 
-    ``APP`` |rarr| ``REQUEST`` |rarr| ``ACTION`` |rarr| ``STEP``
+    ``APP`` |rarr| ``REQUEST`` |rarr| ``ACTION`` |rarr| ``STEP``.
```
Comment (Tishka17) [inline docs/concepts.rst:40]: Why?

Comment (ivan-borovets) [reply docs/concepts.rst:40]: Same in the README. Should we remove it in both places or only here?

Comment (Tishka17) [reply docs/concepts.rst:40]: Formatting features are different. Let's remove trailing dot at least here

Comment (ivan-borovets) [reply docs/concepts.rst:40]: Sure

**Pull request merged**